tasks/base.py

```diff
@@ -29,7 +30,10 @@ def check_response(response: Response, task_name="Task", request=None):
         logger.exception("Non-JSON response.")
         return None, False
 
-    if not str(response.status_code).startswith("2"):
+    if expected_status_code is None:
+        expected_status_code = "2"
```

Comment:
You could initialize the `expected_status_code` param to "2" instead of None, if you really wanted to avoid this check.